### PR TITLE
Add Gargoyle Bonus Type

### PR DESCRIPTION
**Pull request by Toneyisnow**
Adding a bonus type called GARGOYLE according to document:

Unliving creatures always have neutral morale (except Gargoyles). They are composed of Gargoyles, Golems and Elementals. 
All unliving creatures are immune to Destroy Undead, Animate Dead, Resurrection, and Sacrifice. 
They are also immune to the regeneration ability of the Elixir of Life. 
Of creature abilities they are immune to Archangel’s Resurrection, Ghost Dragon's Aging, Mighty Gorgon's Death Stare, Pit Lord's Demon Summoning, Power Lich's Death Cloud (only adjacent splash damage), Vampire Lord's Life Drain, Wyvern Monarch's Poison.
Gargoyles don't have any other immunities or resistances, but Golems and Elementals do.

### CODE REVIEW

File: lib/HeroBonus.h [hunk 1]
```diff
@@ -174,8 +174,9 @@ class DLL_LINKAGE CAddInfo : public std::vector<si32>
 	BONUS_NAME(TWO_HEX_ATTACK_BREATH) /*eg. dragons*/	\
 	BONUS_NAME(SPELL_DAMAGE_REDUCTION) /*eg. golems; value - reduction in %, subtype - spell school; -1 - all, 0 - air, 1 - fire, 2 - water, 3 - earth*/ \
 	BONUS_NAME(NO_WALL_PENALTY)							\
-	BONUS_NAME(NON_LIVING) /*eg. gargoyle*/				\
-	BONUS_NAME(RANDOM_SPELLCASTER) /*eg. master genie, val - level*/ \
+	BONUS_NAME(NON_LIVING) /*eg. golems, cannot be rised or healed, only neutral morale */				\
+    BONUS_NAME(GARGOYLE) /* gargoyle is special than NON_LIVING, cannot be rised or healed */ \
+    BONUS_NAME(RANDOM_SPELLCASTER) /*eg. master genie, val - level*/ \
```
Comment (ArseniyShestakov) [inline lib/HeroBonus.h]: @alexvins is my memory serves me well?
I guess bonuses need to be added to end of list to avoid save breakage.

File: config/creatures/tower.json [hunk 2]
```diff
@@ -392,7 +392,7 @@
 			{
 				"type" : "MIND_IMMUNITY"
 			},
-			"hateArchAngels" : 
+			"hateBlackDragon" :
```
Comment (ArseniyShestakov) [inline config/creatures/tower.json]: Please put changes like that in separate PR. Thanks.

File: lib/HeroBonus.h (hunk 1 above)
Comment (ArseniyShestakov) [reply lib/HeroBonus.h]: And of course it's need to be aligned with tabs not spaces.

Comment (alexvins) [reply lib/HeroBonus.h]: @ArseniyShestakov  You`re correct. Only at end of list

**Comment by ArseniyShestakov** [comment]
Sorry. Fucked up here a bit since reloaded page and continue as Github offered, but really wanted to make squash + merge instead.
